integration-tests/capacity/replenishment.test.ts

```diff
@@ -77,20 +80,12 @@ describe("Capacity Replenishment Testing: ", function () {
 
   describe("Capacity is not replenished", function () {
     it("if out of capacity and last_replenished_at is <= current epoch", async function () {
-      let [stakeKeys, _] = await createAndStakeProvider("NoSend", 150n * CENTS);
+      let [stakeKeys, stakeProviderId] = await createAndStakeProvider("NoSend", 150n * CENTS);
       let payload = JSON.stringify({ changeType: 1, fromId: 1, objectId: 2 })
       let call = ExtrinsicHelper.addOnChainMessage(stakeKeys, schemaId, payload);
 
       // run until we can't afford to send another message.
-      await call.payWithCapacity(-1);
-
-      let nextEpochBlock = await getNextEpochBlock();
-
-      // mine until just before epoch rolls over
-      await ExtrinsicHelper.run_to_block(nextEpochBlock - 1);
-
-      // TODO: it's weird that we can run one more capacity txn before running out.
-      await call.payWithCapacity(-1)
+      await drainCapacity(call, stakeProviderId, stakeKeys);
```

Comment:
Replaced this assumption to a call that will go to a new epoch and drain all capacity.